File: Data Structures & Algorithms/level-order-traversal-of-binary-tree/submission-3.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    private List<List<Integer>> res ;
    public List<List<Integer>> levelOrder(TreeNode root) {
        res = new ArrayList<>();
        dfs(root,0);
        return res;     
    }
    private void dfs(TreeNode n,int d){
        if (n==null){
            return ;
        }
        if(res.size()==d){
            res.add(new ArrayList<>());
        }

        res.get(d).add(n.val);
        dfs(n.left, d + 1);
        dfs(n.right, d + 1);


    }
}

File: Data Structures & Algorithms/path-sum/submission-3.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def hasPathSum(self, root: Optional[TreeNode], targetSum: int) -> bool:
        res=[]
        curr_path=[]
        def dfs(node, curr_sum) :
            if not node :
                return 
            curr_sum+=node.val
            curr_path.append(node.val)

            if curr_sum ==targetSum and not node.left and not node.right:
                res.append(curr_path.copy())
                 
            if node.left :
                dfs(node.left,curr_sum)

            if node.right :
                dfs(node.right,curr_sum)
            curr_path.pop()
        dfs(root,0)
        return len(res)>0
        